modules/os2forms_digital_post/src/Form/SettingsForm.php

```diff
@@ -212,59 +184,43 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    *
    * @phpstan-param array<string, mixed> $form
    */
-  public function validateForm(array &$form, FormStateInterface $formState): void {
-    $triggeringElement = $formState->getTriggeringElement();
-    if ('testCertificate' === ($triggeringElement['#name'] ?? NULL)) {
-      return;
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $config = $this->config(Settings::CONFIG_NAME);
+    foreach ([
+      Settings::TEST_MODE,
+      Settings::SENDER,
+      Settings::CERTIFICATE,
+      Settings::PROCESSING,
+    ] as $key) {
+      $config->set($key, $form_state->getValue($key));
     }
+    $config->save();
 
-    $values = $formState->getValues();
-    if (CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM === $values['certificate']['locator_type']) {
-      $path = $values['certificate'][CertificateLocatorHelper::LOCATOR_TYPE_FILE_SYSTEM]['path'] ?? NULL;
-      if (!file_exists($path)) {
-        $formState->setErrorByName('certificate][file_system][path', $this->t('Invalid certificate path: %path', ['%path' => $path]));
-      }
-    }
+    parent::submitForm($form, $form_state);
   }
 
   /**
-   * {@inheritdoc}
+   * Create form field description with information on any runtime override.
    *
-   * @phpstan-param array<string, mixed> $form
+   * @param string|array $key
+   *   The key.
+   * @param \Drupal\Core\StringTranslation\TranslatableMarkup|null $description
+   *   The actual field description.
+   *
+   * @return string
+   *   The full description.
+   *
+   * @phpstan-param string|string[] $key
    */
-  public function submitForm(array &$form, FormStateInterface $formState): void {
-    $triggeringElement = $formState->getTriggeringElement();
-    if ('testCertificate' === ($triggeringElement['#name'] ?? NULL)) {
-      $this->testCertificate();
-      return;
-    }
-
-    try {
-      $settings['test_mode'] = (bool) $formState->getValue('test_mode');
-      $settings['sender'] = $formState->getValue('sender');
-      $settings['certificate'] = $formState->getValue('certificate');
-      $settings['processing'] = $formState->getValue('processing');
-      $this->settings->setSettings($settings);
-      $this->messenger()->addStatus($this->t('Settings saved'));
-    }
-    catch (OptionsResolverException $exception) {
-      $this->messenger()->addError($this->t('Settings not saved (@message)', ['@message' => $exception->getMessage()]));
+  private function createDescription(string|array $key, ?TranslatableMarkup $description = NULL): string {
```

Comment:
Very good idea 👍

Reply:
Rumor has it that something similar will be added in Drupal 11 core.